### DISCUSSION TITLE
Libre 2 (1 min) deactivation of smoothed values

### DISCUSSION THREAD

**Discussion by Inkalika** [NightscoutFoundation/xDrip#3359] [General]
Hi. I'm using Libre 2+juggluco, 1-minutes values. In xdrip I get as white dots the raw values, which coincide with the values in juggluco, and as blue dots the values smoothed out apparently for 30 minutes. They usually don't match juggluco and the sugar level measured by the glucometer. Because of this, calibrating these smoothed values does not correct the real sugar values. It also causes statistics to be generated from the smoothed values rather than the real values.
How can I disable this smoothing so that the raw values become the main and only ones on the graph?

And another question. Is it possible to transfer only insulin and carbohydrates to NS? This is due to the fact that I transfer blood sugar levels directly from juggluco to NS

![photo_2024-02-24_13-43-53](https://github.com/NightscoutFoundation/xDrip/assets/89869109/f88c39f1-a22a-4861-b7e0-3462c7aa722f)

**Comment by PhilTU** [comment]
I don't believe calibration uses smoothed values, the calibration graph is labelled "raw". But if you turn on data tables in settings and look at the glucose and calibration data tables on the main menu you would be able to validate this. Calibration is more effective when xDrip is connected to the sensor and understands the timing more accurately.

you can reduce the smoothing period see https://xdrip.readthedocs.io/en/latest/install/libre2patch/?h=smoothing#smoothing

What is the reason for using Juggluco ?

Uploading of treatments is optional but I don't see a way to not upload glucose, why not turn that off in juggluco.

**Comment by Inkalika** [comment]
https://xdrip.readthedocs.io/en/latest/install/libre2patch/?h=smoothing#smoothing 
Apparently it's in the new version of the program (my version from 09.23). Thanks, I will look in the Settings menu. Although even 10 minutes smoothing with additional delay of 5 minutes is still inconvenient. Why not give the user the option to opt out of anti-aliasing altogether?

I use Juggluco both because of the fact that I see raw one-minute values and they show up in my widgets. And just because Juggluco can pass raw values to NS and xdrip can't, I pass glucose from Juggluco and with xdrip I pass insulin, carbs and smoothed glucose. I get 2 sugar lines in NS, but it's inconvenient.
So looking for a way out of this situation

**Reply by PhilTU** [reply]
I wasn't clear if the option "If you use filters or calibration plugins, you can select this to send to Nightscout the data as displayed by xDrip+ instead of uncalibrated/unfiltered data." sends raw or calibrated but unsmoothed.

**Reply by Inkalika** [reply]
I couldn't find this option. can you tell me where it is located?

**Reply by PhilTU** [reply]
Nightscout section https://xdrip.readthedocs.io/en/latest/use/cloud/?h=nightscout#nightscout-sync-rest-api
![image](https://github.com/NightscoutFoundation/xDrip/assets/76240903/9ffe1428-c933-46ec-abd3-5d4b09176b95)

**Comment by Inkalika** [comment]
Tell me, where can I disable smoothing in the code for 10, 15, 25 minutes in order to create my own apk without it?
An offset of 5 minutes (when smoothing is selected at 10 minutes) results in insulin being delivered to the AAPS 5 minutes later. But since the readings from the juggluko are not calibrated, I cannot immediately transfer them to the AAPS.

**Reply by Navid200** [reply]
@keencave Can this be disabled?

**Comment by Inkalika** [comment]
I found in the file xDrip/app/src/main/java/com/eveningoutpost/dexdrip/LibreReceiver.java
the code, which I corrected, but I couldn't create an apk in AndroidStudio due to errors during assembly

`
                            case Intents.LIBRE2_BG:
                                Libre2RawValue currentRawValue = processIntent(intent);
                                //JoH.dumpBundle(intent.getExtras(), TAG);

                                if (currentRawValue == null) return;

                                Log.v(TAG, "got bg reading: from sensor:" + currentRawValue.serial + " rawValue:" + currentRawValue.glucose + " at:" + currentRawValue.timestamp);
                                // period of 4.5 minutes to collect 5 readings
                                if (!BgReading.last_within_millis(DexCollectionType.getCurrentDeduplicationPeriod())) {
                                    long smoothing_minutes = Pref.getStringToInt("libre_filter_length", 25);
                                    long dataFetchInterval;
                                    if ( smoothing_minutes == 25L )
                                        dataFetchInterval = 20L;
                                    else
                                        dataFetchInterval = smoothing_minutes;
                                    List<Libre2RawValue> smoothingValues = Libre2RawValue.weightedAverageInterval(dataFetchInterval);
                                    smoothingValues.add(currentRawValue);
                                    processValues(currentRawValue, smoothingValues, smoothing_minutes, context);
                                }
                                currentRawValue.save();
                                clearNFCsensorAge();
                                break;
`

**Reply by Navid200** [reply]
This gives some info on how to do development: https://navid200.github.io/xDrip/docs/dev/dev.html

However, every submission will be scrutinized.  There is no guarantee that every submission will be approved.
Just look at my PRs.  We currently have 64 open PRs.
